File: MyLibrary/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace MyLibrary.Models
{
	[JsonObject(MemberSerialization.OptOut)]
	public class Book
    {
		#region Constructor
		public Book()
		{
		}
		#endregion Constructor

		[Key]
        public int Id { get; set; }

        [Required]
        [StringLength(245)]
        public string Title { get; set; }

        [StringLength(50)]
        public string Author { get; set; }

        //optional
        public int AgeCategoryId { get; set; }
        public int CategoryId { get; set; }
    }
}

File: LendOut/Scripts/app/book.js
System.register([], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var Book;
    return {
        setters: [],
        execute: function () {
            Book = class Book {
                constructor(Id, Title, Author, AgeCategoryId, CategoryId) {
                    this.Id = Id;
                    this.Title = Title;
                    this.Author = Author;
                    this.AgeCategoryId = AgeCategoryId;
                    this.CategoryId = CategoryId;
                }
            };
            exports_1("Book", Book);
        }
    };
});
